### DISCUSSION TITLE
Need help with enabling "Redis cluster" when using Redis Object Cache plugin

### DISCUSSION THREAD

**Discussion by aqeel-tkxel** [rhubarbgroup/.github#28] [Support (ROC)] [answered]
**Scenario:** I have a **Redis cluster on AWS** with **2 shards and 2 read replicas for each.**
So, in total, I see 3 nodes each in the 2 shards; one is master with 2 read replicas for each.

**My current configuration in the wp-config.php are these:**

AWS has given me a single endpoint for the cluster, while nodes have endpoints of their own.
-- My wp-config settings --

```
// adjust Redis host and port if necessary 
define( 'WP_REDIS_PORT', 6379 );

// change the prefix and database for each site to avoid cache data collisions
define( 'WP_REDIS_PREFIX', 'tt-prod-' );
define( 'WP_REDIS_DATABASE', 0 ); // 0-15

// reasonable connection and read+write timeouts
define( 'WP_REDIS_TIMEOUT', 1 );
define( 'WP_REDIS_READ_TIMEOUT', 1 );

// I am not using WP_REDIS_HOST, instead I have:
//Redis Cluster
//https://redis.io/docs/management/scaling/

define( 'WP_REDIS_CLUSTER', [
    'tcp://my-aws-end-point?alias=node-01',
] );
```
-- wp-config ends ----

I want a confirmation if is that is how I need to set the configuration to use the cluster when I have shards and replicas, or do I need to specify both shards endpoints in the WP_REDIS_CLUSTER array, along with any additional configuration?

Also, **FlushAll** command from the Admin panel **doesn't seem to work properly.** Do I need to do any additional configuration when using cluster?

Any help is much appreciated, thank you!

**Comment by tillkruss** [comment, marked as answer]
You only need to specify the cluster's primary nodes. All replicas will be auto-discovered. 

If something isn't working, please post your error logs.

**Reply by aqeel-tkxel** [reply]
Hi,
Thank you for your response:
Below, is the link to the snapshots from my AWS console.

[https://drive.google.com/drive/folders/15lSywkoRMB9ETn93GQHX9J1alDL7Q5hy?usp=sharing](https://drive.google.com/drive/folders/15lSywkoRMB9ETn93GQHX9J1alDL7Q5hy?usp=sharing)

We have 'Cluster mode' enabled, and when we are using that, we use the configuration endpoint (as per the documentation).
[https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/Endpoints.html#Endpoints.Find.RedisCluster](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/Endpoints.html#Endpoints.Find.RedisCluster)
That 'Configuration Endpoint' as per the documentation, (and as seen in redis-cluster.jpg in the link above) is the only index we have in the array in the "WP_REDIS_CLUSTER" variable. Did we do it right?

**Primary nodes are:**

1. cluster-0001-001
2. cluster-0002-001

And **Read replicas are:**

1. cluster-0001-002
2. cluster-0001-003
3. cluster-0002-002
4. cluster-0002-003

**In our case, we only one cluster**.

Please, can you confirm if we have the right settings, by just using "Configuration Endpoint" in "WP_REDIS_CLUSTER", along with the other settings I mentioned above?

And should Flush cache function work as expected with these settings when flushed from the  Wordpress admin panel?

Appreciate your help in this case.

Thank you.

**Reply by tillkruss** [reply]
If you have a cluster with two primary nodes, then yes, list those two in your `WP_REDIS_CLUSTER` array.
